modules/run-vault/run-vault

```diff
@@ -128,13 +135,53 @@ function generate_vault_config {
   local readonly user="$6"
   local readonly gcs_bucket="$7"
   local readonly gcp_creds_file="$8"
+  local readonly enable_auto_unseal="$9"
+  local readonly auto_unseal_project="${10}"
+  local readonly auto_unseal_region="${11}"
+  local readonly auto_unseal_key_ring="${12}"
+  local readonly auto_unseal_crypto_key="${13}"
   local readonly config_path="$config_dir/$VAULT_CONFIG_FILE"
 
   local instance_ip_address
   instance_ip_address=$(get_instance_ip_address)
 
-  log_info "Creating default Vault config file in $config_path"
-  cat > "$config_path" <<EOF
+  if [[ "$enable_auto_unseal" == "true" ]]; then
+    log_info "Creating Vault gcpckms seal config file in $config_path"
+    cat > "$config_path" <<EOF
+storage "gcs" {
+  bucket = "$gcs_bucket"
+  credentials_file = "$gcp_creds_file"
+}
+
+ha_storage "consul" {
+  address = "127.0.0.1:8500"
+  path    = "vault/"
+  scheme  = "http"
+  service = "vault"
+}
+
+seal "gcpckms" {
+  credentials = "$gcp_creds_file"
```

Comment:
is it ok to write this to disk? cant vault infer this from an env var?

Reply:
Yeah, I think its safe, I'm following the lead with the code underneath it as it's not using ENV vars. We don't actually write any secrets to disk, just the location to the credentials file if the customer is using one. They really shouldn't be anyway.